lib/faraday/error.rb

```diff
@@ -85,7 +85,7 @@ def initialize(exc = 'timeout', response = nil)
 
   # Raised by Faraday::Response::RaiseError in case of a nil status in response.
   class NilStatusError < ServerError
-    def initialize(_exc, response: nil)
+    def initialize(_exc, response = nil)
```

Comment:
This is NOT the same thing as `def initialize(_exc, response: nil)`

Reply:
oh, that's true. However, I will point out that this is the same call signature as that of `TimeoutError` on line 80. If this is not the right way, then `TimeoutError` should also be revisited.

Reply:
Can it be rewritten to accept a hash parameter? Like that:
```ruby
def initialize(_exc, params)
  response = params[:response]
```